packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -172,6 +172,54 @@ void main() {
       FileSystem: () => fs,
       ProcessManager: () => mockProcessManager,
     });
+
+    testUsingOsxContext('build settings contains Flutter Xcode environment variables', () async {
+      macOS.environment = <String, String>{
```

Comment:
At runtime the map used by platform.environment will be unmodifiable. For tests that mock the platform, its good to use a `const` map or `Map.unmodifiable` to ensure there is no accidental setting of values, since this will throw during the normal runtime.

Reply:
Good idea.